src/rules/experimental-require-strict-events.ts

```diff
@@ -0,0 +1,45 @@
+import { createRule } from "../utils"
+import { findAttribute, getLangValue } from "../utils/ast-utils"
+
+export default createRule("experimental-require-strict-events", {
+  meta: {
+    docs: {
+      description: "require the strictEvents attribute on <script> tags",
+      category: "Experimental",
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      missingStrictEvents: `The component must have the strictEvents attribute on its <script> tag or it must define the $$Events interface.`,
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    let isTs = false
+    let hasAttribute = false
+    let hasInterface = false
+    return {
+      SvelteScriptElement(node) {
+        const lang = getLangValue(node)?.toLowerCase()
+        isTs = lang === "ts" || lang === "typescript"
+        hasAttribute = findAttribute(node, "strictEvents") !== null
+      },
+      TSInterfaceDeclaration(node) {
+        if (node.id.name === "$$Events") {
+          hasInterface = true
+        }
+      },
+      "Program:exit"() {
+        if (isTs && !hasAttribute && !hasInterface) {
```

Comment:
Do you think the `strictEvents` attribute should be given even if `createEventDispatcher` or `on:click` are not used?

Reply:
Yes, I do - the `strictEvents` attributes makes all "callers" of the component (i.e. components that contain it) have their `on:something` handlers checked as well - so even if a component has no events, the `strictEvents` attribute brings a benefit in that the callers will error if they try to add a handler to a non-existent event.